Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -39,34 +47,71 @@ internal static class ProfileManager
                 }
 
                 int profilingSessionId = GetNextProfilingSessionId();
-                
-                string arguments = System.Environment.ExpandEnvironmentVariables(string.Format("start {0} /attach:{1} /loadAgent:4EA90761-2248-496C-B854-3C0399A591A4;DiagnosticsHub.CpuAgent.dll  /scratchLocation:%LOCAL_EXPANDED%\\Temp", profilingSessionId, processId));
 
-                var profileProcessResponse = await ExecuteProfilingCommandAsync(arguments, tracer);
+                if (iisProfiling)
+                {
+                    var profileResponseStatus = await StartIISSessionAsync(processId, profilingSessionId, tracer);
 
-                if (profileProcessResponse.StatusCode != HttpStatusCode.OK)
+                    if (profileResponseStatus.StatusCode != HttpStatusCode.OK)
```

Comment:
This logic is repeated in both if and else part, move it outside so it is not repeated.